src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -118,19 +178,54 @@ public ReturnValueConfiguration(RuleBuilder parentConfiguration)
 
             public ICallMatcher Matcher => this.ParentConfiguration.Matcher;
 
+            public IReturnValueConfiguration<TMember> Then
+            {
+                get
+                {
+                    var newConfiguration = (RuleBuilder)this.ParentConfiguration.Then;

```

Comment:
A perfectly valid cast in this case, but these downcasts always make me nervous. What's your opinion on removing the cast and making `RuleBuilder.Then` look more like 

```c#
IVoidConfiguration IThenConfiguration<IVoidConfiguration>.Then => this.Then;

private RuleBuilder Then
{
    get
    {
        var newRule = this.RuleBeingBuilt.CloneCallSpecification();
        this.manager.AddRuleAfter(this.RuleBeingBuilt, newRule);
        return new RuleBuilder(newRule, this.manager, this.asserterFactory);
    }
}
```
? Am I just being silly?